src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,134 @@
+            Item milk = new Item("Milk", 29990, ItemType.GADGET)
+            Item appleJuice = new Item("Apple Juice", 174, ItemType.FOOD)
+
+            List<Item> items = [nuggets, ham, milk, appleJuice]
+
+            Cart itemsInTheCart = new Cart(UUID.randomUUID(), items)
+
+        when:
+            boolean hasFoodItems = service.cartContainsFoodItem(itemsInTheCart)
+
+        then:
+            hasFoodItems == true
+    }
+
+    def "calculateTotalCostOfCart should get the sum of totalPrice."() {
+        given:
+            Item nuggets = new Item("Nuggets", 100, ItemType.FOOD)
+            Item ham = new Item("Ham", 100, ItemType.FOOD)
+            Item biscuit = new Item("Biscuit", 50, ItemType.GADGET)
+            Item appleJuice = new Item("Apple Juice", 50, ItemType.FOOD)
+
+            List<Item> items = [nuggets, ham, biscuit, appleJuice]
+
+            Cart itemsInTheCart = new Cart(UUID.randomUUID(), items)
+
+        when:
+            double actualTotalPrice = service.calculateTotalCostOfCart(itemsInTheCart)
+
+        then:
+            300 == actualTotalPrice
+
+    }
+
+    def "isCartEligibleForDiscount should validate if the total price from the cart is eligible for discount."() {
+        given:
+        Item nuggets = new Item("Nuggets", 185.0, ItemType.FOOD)
+        Item ham = new Item("Ham", 198.0, ItemType.FOOD)
+        Item biscuit = new Item("Biscuit", 49.0, ItemType.GADGET)
+        Item appleJuice = new Item("Apple Juice", 174.0, ItemType.FOOD)
+        Item milk = new Item("Milk", 98.0, ItemType.FOOD)
+        Item chocolate = new Item("Chocolate", 357.0, ItemType.FOOD)
+
+        List<Item> items = [nuggets, ham, biscuit, appleJuice, milk, chocolate]
+
+        Cart itemsInTheCart = new Cart(UUID.randomUUID(), items)
+
+        when:
+        boolean eligibleForDiscount = service.isCartEligibleForDiscount(itemsInTheCart)
+
+        then:
+        eligibleForDiscount == true
+
+    }
+
+    def "applyDiscountToCartItems should apply discount when cart is eligible for discount."() {
+        given:
+        Item nuggets = new Item("Nuggets", 50.0, ItemType.FOOD)
+        Item ham = new Item("Ham", 80.0, ItemType.FOOD)
+        Item biscuit = new Item("Biscuit", 50.0, ItemType.GADGET)
+        Item appleJuice = new Item("Apple Juice", 150.0, ItemType.FOOD)
+        Item milk = new Item("Milk", 100.0, ItemType.FOOD)
+        Item chocolate = new Item("Chocolate", 300.0, ItemType.FOOD)
+
+        List<Item> items = [nuggets, ham, biscuit, appleJuice, milk, chocolate]
+
+        Cart itemsInTheCart = new Cart(UUID.randomUUID(), items)
+
+
+        when:
+        service.applyDiscountToCartItems(itemsInTheCart)
+
+        then:
+        5d == items[0].getCost()
+        8d == items[1].getCost()
+        5d == items[2].getCost()
+        15d == items[3].getCost()
+        10d == items[4].getCost()
+        30d == items[5].getCost()
+
+    }
+
+    def "createAnOrder should be able to save the created order."() {
```

Comment:
We should also test the scenario where the `UnableToCreateOrderException` is thrown.